src/attr/_compat.py

```diff
@@ -24,180 +25,76 @@
     ordered_dict = OrderedDict
 
 
-if PY2:
-    from collections import Mapping, Sequence
-
-    from UserDict import IterableUserDict
-
-    # We 'bundle' isclass instead of using inspect as importing inspect is
-    # fairly expensive (order of 10-15 ms for a modern machine in 2016)
-    def isclass(klass):
-        return isinstance(klass, (type, types.ClassType))
-
-    def new_class(name, bases, kwds, exec_body):
-        """
-        A minimal stub of types.new_class that we need for make_class.
-        """
-        ns = {}
-        exec_body(ns)
+def just_warn(*args, **kw):
+    """
+    We only warn on Python 3 because we are not aware of any concrete
+    consequences of not setting the cell on Python 2.
+    """
+    warnings.warn(
+        "Running interpreter doesn't sufficiently support code object "
+        "introspection.  Some features like bare super() or accessing "
+        "__class__ will not work with slotted classes.",
+        RuntimeWarning,
+        stacklevel=2,
+    )
 
-        return type(name, bases, ns)
 
-    # TYPE is used in exceptions, repr(int) is different on Python 2 and 3.
-    TYPE = "type"
+def isclass(klass):
+    return isinstance(klass, type)
```

Comment:
this doesn't really warrant a compat helper anymore, but that can be removed later

Reply:
i literally wrote about that in the PR ;)

i want to keep it as short as possible, there'll be more subsequent cleanups